flake: TestAPIConversion

```
=== RUN   TestAPIConversion
=== PAUSE TestAPIConversion
=== CONT  TestAPIConversion
    conversion_test.go:51: shared kcp server will target configuration "/go/src/github.com/kcp-dev/kcp/.kcp/admin.kubeconfig"
    conversion_test.go:61: Created root:organization workspace root:e2e-workspace-6cmfb as /clusters/1zd3izido3ierzx6
    conversion_test.go:69: Setting up APIResourceSchema, APIConversion, APIExport, and APIBinding for widgets
I0118 02:36:03.995109   30995 bootstrap.go:236] "upserted object" apiresourceschema.workspace="1zd3izido3ierzx6" apiresourceschema.namespace="" apiresourceschema.name="rev0002.widgets.example.io" apiresourceschema.apiVersion="apis.kcp.io/v1alpha1"
I0118 02:36:04.006283   30995 bootstrap.go:236] "upserted object" apiconversion.workspace="1zd3izido3ierzx6" apiconversion.namespace="" apiconversion.name="rev0002.widgets.example.io" apiconversion.apiVersion="apis.kcp.io/v1alpha1"
I0118 02:36:04.014902   30995 bootstrap.go:236] "upserted object" apiexport.workspace="1zd3izido3ierzx6" apiexport.namespace="" apiexport.name="widgets.example.io" apiexport.apiVersion="apis.kcp.io/v1alpha1"
I0118 02:36:04.025098   30995 bootstrap.go:236] "upserted object" apibinding.workspace="1zd3izido3ierzx6" apibinding.namespace="" apibinding.name="widgets.example.io" apibinding.apiVersion="apis.kcp.io/v1alpha1"
    conversion_test.go:78: Waiting for initial binding to complete
    conversion_test.go:90: Resetting the RESTMapper so it can pick up widgets
    conversion_test.go:93: Creating v1 widget
    conversion_test.go:95: 
        	Error Trace:	conversion_test.go:95
        	Error:      	Received unexpected error:
        	            	failed to create resource v1-widget.yaml doc 1: could not get REST mapping for example.io/v1, Kind=Widget: no matches for kind "Widget" in version "example.io/v1"
        	Test:       	TestAPIConversion
        	Messages:   	error creating v1 widget
--- FAIL: TestAPIConversion (1.29s)
```

From https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2638/pull-ci-kcp-dev-kcp-main-e2e-sharded/1615532489940406272

Probably need to do an `Eventually` loop that keeps resetting the RESTMapper and then tries to create the Widget.

--- a/test/e2e/conversion/conversion_test.go
+++ b/test/e2e/conversion/conversion_test.go
@@ -87,12 +87,17 @@ func TestAPIConversion(t *testing.T) {
 		return false, fmt.Sprintf("%v", apiBinding.Status)
 	}, wait.ForeverTestTimeout, 100*time.Millisecond, "APIBinding never completed its initial binding")
 
-	t.Logf("Resetting the RESTMapper so it can pick up widgets")
-	mapper.Reset()
+	framework.Eventually(t, func() (success bool, reason string) {
+		t.Logf("Resetting the RESTMapper so it can pick up widgets")
+		mapper.Reset()
 
-	t.Logf("Creating v1 widget")
-	err = helpers.CreateResourceFromFS(ctx, dynamicClusterClient.Cluster(orgPath), mapper, nil, "v1-widget.yaml", embeddedResources)
-	require.NoError(t, err, "error creating v1 widget")
+		t.Logf("Creating v1 widget")
+		err = helpers.CreateResourceFromFS(ctx, dynamicClusterClient.Cluster(orgPath), mapper, nil, "v1-widget.yaml", embeddedResources)
+		if err == nil {
+			return true, ""
+		}
+		return false, err.Error()
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "Unable to create Widget")
 
 	v2GVR := schema.GroupVersionResource{Group: "example.io", Version: "v2", Resource: "widgets"}
 

